storm-core/src/jvm/org/apache/storm/metric/api/IMetricsConsumer.java

```diff
@@ -54,6 +54,25 @@ public String toString() {
         }
         public String name; 
         public Object value;
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (!(o instanceof DataPoint)) return false;
+
+            DataPoint dataPoint = (DataPoint) o;
+
+            if (name != null ? !name.equals(dataPoint.name) : dataPoint.name != null) return false;
```

Comment:
This is IntelliJ generated code so might not beauty and doesn't rely on other library. StringUtils.equals() would be better. I'll address it.